Release notes for Mimir 2.2. (#2221)
grafana/mimir: docs/sources/release-notes/v2.2.md

---

> title: "Grafana Mimir version 2.1 release notes"

Reviewer: ```suggestion
title: "Grafana Mimir version 2.2 release notes"
```

menuTitle: "V2.2 release notes"
description: "Release notes for Grafana Mimir version 2.2"
weight: 300
---

# Grafana Mimir version 2.2 release notes

Grafana Labs is excited to announce version 2.2 of Grafana Mimir, the most scalable, most performant open source time series database in the world.

Highlights include the top features, enhancements, and bugfixes in this release. If you are upgrading from Grafana Mimir 2.1, there is migration-related information as well.
For the complete list of changes, see the [Changelog](https://github.com/grafana/mimir/blob/main/CHANGELOG.md).

> ## Features and enhancements

Reviewer 2: I would call out the Helm chart too. It's true it has its own release cycle and changelog, but we should call it out as a massive improvement since Mimir 2.1.0. @krajorama and @Logiraptor could help drafting the notes.

Author: Added placeholder for now:

> - **Helm Chart update**: TBD

> ## Features and enhancements

Reviewer 2: Faster WAL replay times looks a very nice improvement too. Do we have numbers to share?

```
* [ENHANCEMENT] Ingester: reduce sleep time when reading WAL. #2098
```

Author: From our internal monitoring cluster, it went from ~7 minutes on average to ~3 minutes. We can say it went down by about 50% percent.

Author: Added:

> - **Faster ingester startup**: Ingesters now replay Write-Ahead-Log by about 50% faster, and they also re-join the ring sooner under some conditions.

> ## Features and enhancements

Author: It's such a low-level change, that I have difficulty phrasing it. What about:

> Store-gateway can now optionally prepopulate the file system cache when memory-mapping index-header files. This can help store-gateway to avoid looking stuck while processing index-headers. Feature can be enabled with new experimental flag `-blocks-storage.bucket-store.index-header.map-populate-enabled`.

> ## Features and enhancements

Reviewer 2: LGTM, I would just do the following change:

> while processing index-headers
while loading index-headers

> It's such a low-level change
True it's low level, but impactful.

> ## Features and enhancements

Author: > > It's such a low-level change
> True it's low level, but impactful.

I didn't mean to downplay it, I was just wondering how to explain it :)

> - **Support for ingesting out-of-order samples**: Grafana Mimir introduces new experimental support for ingesting out-of-order samples. Feature is enabled by configuring the time window since the most-recently ingested sample for the series. This can be enabled on per-tenant basis.

Author: Documentation is being prepared in https://github.com/grafana/mimir/pull/2183, it seems it will be ready before 2.2 release.

> - **Support for ingesting out-of-order samples**: Grafana Mimir introduces new experimental support for ingesting out-of-order samples. Feature is enabled by configuring the time window since the most-recently ingested sample for the series. This can be enabled on per-tenant basis.

Author: Added TODO so that we don't forget.

  This support is configurable, with users able to set how far out-of-order Mimir will accept samples on a per-tenant basis.
  Note that this feature still needs a heavy testing, and is not production-ready yet.

>   **TODO: Link to documentation after https://github.com/grafana/mimir/pull/2183 is merged.**

Reviewer 2: PR is merged now ;)

Author: Added link.

> - **Error messages**: We have overhauled the error messages reported to end user to make them easier to understand, and to include error codes that can be easily searched for. This effort will continue in next releases as well.

Author: Added link to operators-guide/mimir-runbooks/_index.md#codified-errors.

> - **Error messages**: The error messages that Mimir reports are more human readable, and the messages include error codes that are easily searchable.

Reviewer 2: Let's not call it "codified errors" but "errors catalog". @zenador can you open a PR to rename it in `mimir-runbooks/_index.md`, please? Then here in the release note we could say we introduced a Mimir errors catalog.

> - **Prefixed bucket store**: It is now possible to store blocks, rules and alerts into a single bucket, by configuring bucket prefix for each feature.

Reviewer: ```suggestion
- **Configurable prefix for object storage**: Mimir can now store block data, rules, and alerts in one bucket, each under its own user-defined prefix, rather than requiring one bucket for each.
```

Reviewer: Curious - as a GEM customer, can I set a bucket prefix for the admin-api storage? (Wouldn't be in scope for these release notes, but we'll want to make this callout in the GEM release notes if this is the case). 

Can we point to the section in the configuration this is? or at least mention the name of the right configuration flag?

Author: > Curious - as a GEM customer, can I set a bucket prefix for the admin-api storage?

@dimitarvdimitrov may know the answer.

> - **Backfill API**: Grafana Mimir 2.2 introduces new API for backfill. Work on client is in progress. (Shall we drop this until client is available?)

Reviewer: I can't tell from this if we should include this. Putting myself in the shoes of a user, is the backfill api 'ready enough' that I can achieve the main outcome that I want - namely the ability to import historic data into Mimir from my Prometheus? If we can do this, we should DEFINITELY include this and we should put it at the top of the list, under the out-of-order support. This is a really awesome feature!

Reviewer 3: I hope we can finish mimirtool before the end of this week. I will prioritize it :)

## Upgrade considerations

We have updated default values and some parameters in Grafana Mimir 2.2 to give users better out-of-the-box experience:

> - Message size limits for gRPC messages exchanged between internal Mimir components has been increased to 100 MiB.

Reviewer: How does this help me? Fewer messages get dropped? In general, our average message is bigger than the default message size used to satisfy?

> - Message size limits for gRPC messages exchanged between internal Mimir components has been increased to 100 MiB.

Author: Expanded entry, ptal.

> - `-blocks-storage.bucket-store.ignore-blocks-within` parameter has been changed from `0` to `10h`. Default value of `-querier.query-store-after` has been changed from 0 to `12h`. Both changes improve query-performance for most-recent data by querying only ingesters.

Reviewer: ```suggestion
- The `-blocks-storage.bucket-store.ignore-blocks-within` parameter has been changed from `0` to `10h`. The default value of `-querier.query-store-after` has been changed from `0` to `12h`. Both changes improve query-performance for most-recent data by querying only the ingesters, rather than object storage.
```

> - Ingester's `-ingester.ring.join-after` is no longer respected, and ingester treats it as 0.

Reviewer: What does this do for the user? Does this mean that this flag is deprecated?

Author: Hmm, it doesn't help users which never changed it -- it defaulted to 0s already. I think we should remove this entry.

> - If you use shuffle sharding and configure lookback period in queriers and ruler, note that option `-querier.shuffle-sharding-ingesters-lookback-period` has been deprecated, and replaced with `-querier.shuffle-sharding-ingesters-enabled`. Lookback period now equals to value specified by `-querier.query-ingesters-within` option.

Reviewer: ```suggestion
- If you use shuffle sharding and configure a lookback period in the queriers and the ruler, note that option `-querier.shuffle-sharding-ingesters-lookback-period` has been deprecated, and replaced with `-querier.shuffle-sharding-ingesters-enabled`. The lookback period now equals to value specified by `-querier.query-ingesters-within` option.
```

Hmm...is another way to put it...
The option `-querier.shuffle-sharding-ingesters-lookback-period` has been deprecated. Instead, users of shuffle sharding who want to enable querying the ingesters should set `-querier.shuffle-sharding-ingesters-enabled` to `true` and specify the lookback period by setting the `-querier.query-ingesters-within` option. For those using shuffle sharding with `-querier.shuffle-sharding-ingesters-enabled` set to false, all queries go to object storage, rather than to the ingesters.

Reviewer 4: > For those using shuffle sharding with -querier.shuffle-sharding-ingesters-enabled set to false, all queries go to object storage, rather than to the ingesters.

This isn't quite correct, with `-querier.shuffle-sharding-ingesters-enabled=false` all queries within `-querier.query-ingesters-within` go to _all ingesters_ instead of just a subset of them.

Reviewer: @56quarters -- better?

The option -querier.shuffle-sharding-ingesters-lookback-period has been deprecated. Anyone who previously changed this option from its default of `0s` should instead set `-querier.shuffle-sharding-ingesters-enabled` to true and specify the lookback period by setting the` -querier.query-ingesters-within` option.

Reviewer 4: Yep!

Author: Using the latest version from https://github.com/grafana/mimir/pull/2221#discussion_r906307778. Thank you both!

> - When using memberlist and component fails to join the memberlist cluster, it no longer aborts startup by default: `-memberlist.abort-if-join-fails` parameter now defaults to false.

Reviewer: ```suggestion
- The `-memberlist.abort-if-join-fails` parameter now defaults to false. This means that when using memberlist and a component fails to join the memberlist cluster, it no longer aborts startup by default.
```

## Bug fixes

> - [PR 1883](https://github.com/grafana/mimir/pull/1883): Fixed a bug that caused Grafana Mimir to crash when it received user-query with special regular expression label matcher.

Reviewer: ```suggestion
- [PR 1883](https://github.com/grafana/mimir/pull/1883): Fixed a bug that caused Grafana Mimir to crash when it received a user query with a special regular expression label matcher.
```

> ## Bug fixes

Reviewer 2: I would also mention:

```
* [BUGFIX] Fix panic at startup when Mimir is running in monolithic mode and query sharding is enabled. #2036
```

> - [PR 1933](https://github.com/grafana/mimir/pull/1933): Fixed a bug in ring page exposed by Ingesters, which showed incorrect status of entries in the ring.

Reviewer: ```suggestion
- [PR 1933](https://github.com/grafana/mimir/pull/1933): Fixed a bug in the [ingester ring page](link), which showed incorrect status of entries in the ring.
```

I think we should link to https://grafana.com/docs/mimir/next/operators-guide/reference-http-api/#ingesters-ring-status

> - [PR 2090](https://github.com/grafana/mimir/pull/2090): When ruler uses remote rule evaluation, it now applies timeout correctly. Previously it could get stuck forever and stop rule evaluation.

Reviewer: ```suggestion
- [PR 2090](https://github.com/grafana/mimir/pull/2090): Ruler in [remote rule evaluation mode](link) now applies timeout correctly. Previously it could get stuck forever, effectively halting rule evaluation.
```

Link to https://grafana.com/docs/mimir/next/operators-guide/architecture/components/ruler/#remote

---
title: "Grafana Mimir version 2.2 release notes"
menuTitle: "V2.2 release notes"
description: "Release notes for Grafana Mimir version 2.2"
weight: 300
---

# Grafana Mimir version 2.2 release notes

Grafana Labs is excited to announce version 2.2 of Grafana Mimir, the most scalable, most performant open source time series database in the world.

> Below we highlight the top features, enhancements and bugfixes in this release, as well as relevant callouts for those upgrading from Grafana Mimir 2.1.
> The complete list of changes is recorded in the [Changelog](https://github.com/grafana/mimir/blob/main/CHANGELOG.md).

Reviewer 5: ```suggestion
Highlights include the top features, enhancements, and bugfixes in this release. If you are upgrading from Grafana Mimir 2.1, there is migration-related information as well.
For the complete list of changes, see the [Changelog](https://github.com/grafana/mimir/blob/main/CHANGELOG.md).
```

## Features and enhancements

> - **Support for ingesting out-of-order samples**: Grafana Mimir introduces new experimental support for ingesting out-of-order samples.

Reviewer 5: ```suggestion
- **Support for ingesting out-of-order samples**: Grafana Mimir includes new, experimental support for ingesting out-of-order samples.
```

  This support is configurable, with users able to set how far out-of-order Mimir will accept samples on a per-tenant basis.
  **TODO: Link to documentation after https://github.com/grafana/mimir/pull/2183 is merged.**

- **Backfill API**: Grafana Mimir 2.2 introduces new API for backfill. TODO: expand once mimirtool changes are in.

> - **Error messages**: We have made the error messages reported by Mimir more human readable and included error codes that can be easily searched for.
>   These error message improvements will continue in future releases.
>   You can find description of errors in [Grafana Mimir Runbooks / Codified errors]({{< relref "../operators-guide/mimir-runbooks/_index.md#codified-errors" >}}).

Reviewer 5: ```suggestion
- **Error messages**: The error messages that Mimir reports are more human readable, and the messages include error codes that are easily searchable.
  For descriptions of errors, see the [Grafana Mimir runbooks / Codified errors]({{< relref "../operators-guide/mimir-runbooks/_index.md#codified-errors" >}}).
```
Remove future-oriented statements. Simply include them in the next set of release notes. Baked-in promises can backfire.

> - **Configurable prefix for object storage**: Mimir can now store block data, rules, and alerts in one bucket, each under its own user-defined prefix, rather than requiring one bucket for each. Storage prefix can be configured using `-<storage>.storage-prefix` option for corresponding storage: `ruler-storage`, `alertmanager-storage` or `blocks-storage`.

Reviewer 5: ```suggestion
- **Configurable prefix for object storage**: Mimir can now store block data, rules, and alerts in one bucket, each under its own user-defined prefix, rather than requiring one bucket for each. You can configure the storage prefix by using `-<storage>.storage-prefix` option for corresponding storage: `ruler-storage`, `alertmanager-storage` or `blocks-storage`.
```
Change passive voice, where the doer is known, to active voice. This allows you to address the reader directly.

Reviewer 5: Look out for any verbiage that has the construction, "is ...-ed" or "can be ... -ed", and you will often find passive voice.

## Upgrade considerations

> We have updated default values and some parameters in Grafana Mimir 2.2 to give users better out-of-the-box experience:

Reviewer 5: ```suggestion
We have updated default values and some parameters in Grafana Mimir 2.2 to give you a better out-of-the-box experience:
```

> - Message size limits for gRPC messages exchanged between internal Mimir components has been increased to 100 MiB from previous 4 MiB. This helps to avoid internal server errors when pushing or querying large data.

Reviewer 5: ```suggestion
- Message size limits for gRPC messages exchanged between internal Mimir components increased to 100 MiB from the previous 4 MiB. This helps to avoid internal server errors when pushing or querying large data.
```

> - The `-blocks-storage.bucket-store.ignore-blocks-within` parameter has been changed from `0` to `10h`. The default value of `-querier.query-store-after` has been changed from `0` to `12h`. Both changes improve query-performance for most-recent data by querying only the ingesters, rather than object storage.

Reviewer 5: ```suggestion
- The `-blocks-storage.bucket-store.ignore-blocks-within` parameter changed from `0` to `10h`. The default value of `-querier.query-store-after` changed from `0` to `12h`. Both changes improve query-performance for most-recent data by querying only the ingesters, rather than object storage.
```

> The option `-querier.shuffle-sharding-ingesters-lookback-period` has been deprecated. Anyone who previously changed this option from its default of `0s` should instead set `-querier.shuffle-sharding-ingesters-enabled` to `true` and specify the lookback period by setting the `-querier.query-ingesters-within` option.

Reviewer 5: ```suggestion
The option `-querier.shuffle-sharding-ingesters-lookback-period` has been deprecated. If you previously changed this option from its default of `0s`, set `-querier.shuffle-sharding-ingesters-enabled` to `true` and specify the lookback period by setting the `-querier.query-ingesters-within` option.
```

> - The `-memberlist.abort-if-join-fails` parameter now defaults to false. This means that when using memberlist and a component fails to join the memberlist cluster, it no longer aborts startup by default.

Reviewer 5: Clarify what "it" is in the phrase,  "it no longer aborts startup by default."

Grammatically, "it" can mean the component, the parameter, or the cluster, or even possibly some assumed knowledge that is implicit.

Author: "it" here refers to Mimir component. WDYT about this instead?

```suggestion
- The `-memberlist.abort-if-join-fails` parameter now defaults to false. This means that when using memberlist and a Mimir fails to join the memberlist cluster, Mimir no longer aborts startup by default.
```

Author: I've used this for now:

> - The `-memberlist.abort-if-join-fails` parameter now defaults to false. When Mimir is using memberlist as a backend store for hash ring, and it fails to join the memberlist cluster, Mimir no longer aborts startup by default.

## Bug fixes

> - [PR 1883](https://github.com/grafana/mimir/pull/1883): Fixed a bug that caused Grafana Mimir to crash when it received a user query with a special regular expression label matcher.

Reviewer 5: ```suggestion
- [PR 1883](https://github.com/grafana/mimir/pull/1883): Fixed a bug that caused the query-frontend and querier to crash when they received a user query with a special regular expression label matcher.
```

Reviewer 5: Reduced the scope to not all of Grafana Mimir, with @pracucci’s input.

- [PR 1933](https://github.com/grafana/mimir/pull/1933): Fixed a bug in the [ingester ring page]({{< relref "../operators-guide/reference-http-api/index.md#ingesters-ring-status" >}}), which showed incorrect status of entries in the ring.

> - [PR 2090](https://github.com/grafana/mimir/pull/2090): Ruler in [remote rule evaluation mode]({{< relref "../operators-guide/architecture/components/ruler/index.md#remote" >}}) now applies timeout correctly. Previously the ruler could get stuck forever, effectively halting rule evaluation.

Reviewer 5: ```suggestion
- [PR 2090](https://github.com/grafana/mimir/pull/2090): Ruler in [remote rule evaluation mode]({{< relref "../operators-guide/architecture/components/ruler/index.md#remote" >}}) now applies the timeout correctly. Previously the ruler could get stuck forever, which halted rule evaluation.
```

---
title: "Grafana Mimir version 2.2 release notes"
menuTitle: "V2.2 release notes"
description: "Release notes for Grafana Mimir version 2.2"
weight: 300
---

# Grafana Mimir version 2.2 release notes

Grafana Labs is excited to announce version 2.2 of Grafana Mimir, the most scalable, most performant open source time series database in the world.

> Highlights include the top features, enhancements, and bugfixes in this release. If you are upgrading from Grafana Mimir 2.1, there is migration-related information as well.

Reviewer 6: Should this be `upgrade-related`? cc @osg-grafana since @09jvilla is not available.

> Highlights include the top features, enhancements, and bugfixes in this release. If you are upgrading from Grafana Mimir 2.1, there is migration-related information as well.

Reviewer 6: Edited @09jvilla's suggestion before applying. Did `s/upgraded-related/upgrade-related/`

  This support is configurable, with users able to set how far out-of-order Mimir will accept samples on a per-tenant basis.
  Note that this feature still needs a heavy testing, and is not production-ready yet.
  For documentation see [Configuring out-of-order samples ingestion]({{< relref "../operators-guide/configuring/configuring-out-of-order-samples-ingestion.md" >}})

- **Error messages**: The error messages that Mimir reports are more human readable, and the messages include error codes that are easily searchable.
  For descriptions of errors, see the [Grafana Mimir runbooks / Errors catalog]({{< relref "../operators-guide/mimir-runbooks/_index.md#errors-catalog" >}}).

- **Configurable prefix for object storage**: Mimir can now store block data, rules, and alerts in one bucket, each under its own user-defined prefix, rather than requiring one bucket for each.
  You can configure the storage prefix by using `-<storage>.storage-prefix` option for corresponding storage: `ruler-storage`, `alertmanager-storage` or `blocks-storage`.

- **Helm Chart update**: TBD

> - **Store-gateway can now optionally prepopulate the file system cache when memory-mapping index-header files.**

Reviewer: ```suggestion
- **Store-gateway performance optimization**
```

>   This can help store-gateway to avoid looking stuck while loading index-headers.

Reviewer: ```suggestion
  The store-gateway can now pre-populate the file system cache when memory-mapping index-header files. This can help the store-gateway avoid looking stuck while loading [index-headers]({{< relref "../operators-guide/architecture/binary-index-header" >}}). This feature is experimental and disabled by default; enable it using the flag `-blocks-storage.bucket-store.index-header.map-populate-enabled`.
```

>   This can help store-gateway to avoid looking stuck while loading index-headers.
>   Feature can be enabled with new experimental flag `-blocks-storage.bucket-store.index-header.map-populate-enabled`.

Reviewer 6: ```suggestion
  The store-gateway can now pre-populate the file system cache when memory-mapping index-header files.
  This can help the store-gateway avoid looking stuck while loading [index-headers]({{< relref "../operators-guide/architecture/binary-index-header" >}}). 
  This feature is experimental and disabled by default; enable it using the flag `-blocks-storage.bucket-store.index-header.map-populate-enabled`.
```

@09jvilla's suggestion from above applied to both lines that have to be replaced.

- **Faster ingester startup**: Ingesters now replay Write-Ahead-Log by about 50% faster, and they also re-join the ring sooner under some conditions.

## Upgrade considerations

We have updated default values and some parameters in Grafana Mimir 2.2 to give you a better out-of-the-box experience:

- Message size limits for gRPC messages exchanged between internal Mimir components increased to 100 MiB from the previous 4 MiB.
  This helps to avoid internal server errors when pushing or querying large data.

- The `-blocks-storage.bucket-store.ignore-blocks-within` parameter changed from `0` to `10h`.
  The default value of `-querier.query-store-after` changed from `0` to `12h`.
  Both changes improve query performance for most-recent data by querying only the ingesters, rather than object storage.

- The option `-querier.shuffle-sharding-ingesters-lookback-period` has been deprecated.
  If you previously changed this option from its default of `0s`, set `-querier.shuffle-sharding-ingesters-enabled` to `true` and specify the lookback period by setting the `-querier.query-ingesters-within` option.

- The `-memberlist.abort-if-join-fails` parameter now defaults to false.

>   When Mimir is using memberlist as a backend store for hash ring, and it fails to join the memberlist cluster, Mimir no longer aborts startup by default.

Reviewer: ```suggestion
  When Mimir is using memberlist as the backend store for its hash ring, and it fails to join the memberlist cluster, Mimir no longer aborts startup by default.
```

---
title: "Grafana Mimir version 2.2 release notes"
menuTitle: "V2.2 release notes"
description: "Release notes for Grafana Mimir version 2.2"
weight: 300
---

# Grafana Mimir version 2.2 release notes

Grafana Labs is excited to announce version 2.2 of Grafana Mimir, the most scalable, most performant open source time series database in the world.

Highlights include the top features, enhancements, and bugfixes in this release. If you are upgrading from Grafana Mimir 2.1, there is migration-related information as well.
For the complete list of changes, see the [Changelog](https://github.com/grafana/mimir/blob/main/CHANGELOG.md).

## Features and enhancements

- **Support for ingesting out-of-order samples**: Grafana Mimir includes new, experimental support for ingesting out-of-order samples.

>   This support is configurable, with users able to set how far out-of-order Mimir will accept samples on a per-tenant basis.

Reviewer 5: ```suggestion
  This support is configurable, and it allows you to set how far out-of-order Mimir accepts samples on a per-tenant basis.
```

>   This feature still needs additional testing; we do not recommend using it in production settings yet.

Reviewer 5: ```suggestion
  This feature still needs additional testing; we do not recommend using it in a production environment.
```

>   For documentation see [Configuring out-of-order samples ingestion]({{< relref "../operators-guide/configuring/configuring-out-of-order-samples-ingestion.md" >}})

Reviewer 5: ```suggestion
  For more information, see [Configuring out-of-order samples ingestion]({{< relref "../operators-guide/configuring/configuring-out-of-order-samples-ingestion.md" >}})
```

- **Improved error messages**: The error messages that Mimir reports are more human readable, and the messages include error codes that are easily searchable.

>   For descriptions of errors, see the [Grafana Mimir runbooks / Errors catalog]({{< relref "../operators-guide/mimir-runbooks/_index.md#errors-catalog" >}}).

Reviewer 5: ```suggestion
  For error descriptions, see the Grafana Mimir runbooks’ [Errors catalog]({{< relref "../operators-guide/mimir-runbooks/_index.md#errors-catalog" >}}).
```

> - **Configurable prefix for object storage**: Mimir can now store block data, rules, and alerts in one bucket, each under its own user-defined prefix, rather than requiring one bucket for each.
>   You can configure the storage prefix by using `-<storage>.storage-prefix` option for corresponding storage: `ruler-storage`, `alertmanager-storage` or `blocks-storage`.

Reviewer 5: ```suggestion
- **Configurable prefix for object storage**: Mimir can now store block data, rules, and alerts in one bucket, with each under its own user-defined prefix, rather than requiring one bucket for each.
  You can configure the storage prefix by using `-<storage>.storage-prefix` option for corresponding storage: `ruler-storage`, `alertmanager-storage` or `blocks-storage`.
```

- **Store-gateway performance optimization**
  The store-gateway can now pre-populate the file system cache when memory-mapping index-header files.

>   This can help the store-gateway avoid looking stuck while loading [index-headers]({{< relref "../operators-guide/architecture/binary-index-header" >}}).

Reviewer 5: ```suggestion
  This avoid the store-gateway from appearing to be stuck while loading [index-headers]({{< relref "../operators-guide/architecture/binary-index-header" >}}).
```

- **Helm Chart improvements**: The [Mimir Helm chart](https://github.com/grafana/mimir/tree/main/operations/helm/charts/mimir-distributed) is the best way to install Mimir on Kubernetes. As part of the Mimir 2.2 release, we're also releasing version 3.0 of the Helm chart. Notable enhancements follow. For the full list of changes, see the [Helm chart changelog](https://github.com/grafana/mimir/tree/main/operations/helm/charts/mimir-distributed/CHANGELOG.md).
  - The Helm chart now supports [OpenShift](https://www.redhat.com/en/technologies/cloud-computing/openshift).
  - We have significantly improved the configuration experience for the Helm chart. The most salient one is that we've added a `structuredConfig` option, which allows you to overwrite specific key-value pairs in the `mimir.config` template, which saves you from having to maintain the entire `mimir.config` in your own `values.yaml` file.
  - We've improved the documentation for the Helm chart by adding a [Getting started with Mimir using the Helm chart]({{< relref "../operators-guide/deploying-grafana-mimir/getting-started-helm-charts/_index.md" >}}).

## Upgrade considerations

> We have updated default values and some parameters in Grafana Mimir 2.2 to give you a better out-of-the-box experience:

Reviewer 5: ```suggestion
In Grafana Mimir 2.2, we have updated default values and some parameters to give you a better out-of-the-box experience:
```

> - Message size limits for gRPC messages exchanged between internal Mimir components increased to 100 MiB from the previous 4 MiB.

Reviewer 5: ```suggestion
- Message size limits for gRPC messages that are exchanged between internal Mimir components have increased to 100 MiB from 4 MiB.
```

  This helps to avoid internal server errors when pushing or querying large data.

- The `-blocks-storage.bucket-store.ignore-blocks-within` parameter changed from `0` to `10h`.
  The default value of `-querier.query-store-after` changed from `0` to `12h`.

>   Both changes improve query performance for most-recent data by querying only the ingesters, rather than object storage.

Reviewer 5: ```suggestion
  For most-recent data, both changes improve query performance by querying only the ingesters, rather than object storage.
```

- The option `-querier.shuffle-sharding-ingesters-lookback-period` has been deprecated.
  If you previously changed this option from its default of `0s`, set `-querier.shuffle-sharding-ingesters-enabled` to `true` and specify the lookback period by setting the `-querier.query-ingesters-within` option.

> - The `-memberlist.abort-if-join-fails` parameter now defaults to false.

Reviewer 5: ```suggestion
- The `-memberlist.abort-if-join-fails` parameter now defaults to `false`.
```

  When Mimir is using memberlist as the backend store for its hash ring, and it fails to join the memberlist cluster, Mimir no longer aborts startup by default.

> If you have used a previous version of the [Mimir Helm chart](https://github.com/grafana/mimir/tree/main/operations/helm/charts/mimir-distributed), you must address some of the chart's breaking changes before upgrading to helm chart version 3.0. For a detailed information about to do this, see [Upgrade the Grafana Mimir Helm chart from version 2.1 to 3.0]({{< relref "../operators-guide/deploying-grafana-mimir/upgrade-helm-chart-2.1-to-3.0/_index.md" >}}).

Reviewer 5: ```suggestion
If you have used a previous version of the [Mimir Helm chart](https://github.com/grafana/mimir/tree/main/operations/helm/charts/mimir-distributed), you must address some of the chart's breaking changes before upgrading to helm chart version 3.0. For a detailed information about how to do this, see [Upgrade the Grafana Mimir Helm chart from version 2.1 to 3.0]({{< relref "../operators-guide/deploying-grafana-mimir/upgrade-helm-chart-2.1-to-3.0/_index.md" >}}).
```

    - We've added a `structuredConfig` option, which allows you to overwrite specific key-value pairs in the `mimir.config` template, which saves you from having to maintain the entire `mimir.config` in your own `values.yaml` file.
    - We've added the ability to create global pod annotations. This unlocks the ability to trigger a restart of all services in response to a single event, such as the update of the secret containing Mimir's storage credentials.
  - We've set the chart to disable `-ingester.ring.unregister-on-shutdown` and `-distributor.extend-writes`, for a smoother upgrade experience. Rolling restarts of ingesters are now less likely to cause spikes in resource usage.
  - We've improved the documentation for the Helm chart by adding a [Getting started with Mimir using the Helm chart]({{< relref "../operators-guide/deploying-grafana-mimir/getting-started-helm-charts/_index.md" >}}).

>   - We've added a smoke test for your Mimir cluster to help catch errors immediately after you install or upgrade Mimir via the Helm chart.

Author: That's a lot of changes to point out in this section in my opinion. We still refer people to the upgrading instructions and full changelog, so should we perhaps only pick most important changes here?

/cc @dimitarvdimitrov

Reviewer 7: fair point, it's taking about a 1/3 of the full release notes. I will trim it down
